river/tree/nodes/sgt_nodes.py

```diff
@@ -0,0 +1,223 @@
+                left_dlms = stats.Var()
+                for thresh, ghs in quantizer:
+                    left_ghs += ghs
+                    left_delta_pred = self.delta_prediction(
+                        left_ghs.mean, sgt.lambda_value
+                    )
+                    left_dlms += left_ghs.delta_loss_mean_var(left_delta_pred)
+
+                    right_ghs = self._update_stats - left_ghs
+                    right_delta_pred = self.delta_prediction(
+                        right_ghs.mean, sgt.lambda_value
+                    )
+                    right_dlms = right_ghs.delta_loss_mean_var(right_delta_pred)
+
+                    all_dlms = left_dlms + right_dlms
+
+                    loss_mean = all_dlms.mean.get()
+                    loss_var = all_dlms.get()
+
+                    if loss_mean < candidate.merit.loss_mean:
+                        candidate.merit.loss_mean = (
+                            loss_mean + 2.0 * sgt.gamma / self.total_weight
+                        )
+                        candidate.merit.loss_var = loss_var
+                        candidate.merit.delta_pred[0] = left_delta_pred
+                        candidate.merit.delta_pred[1] = right_delta_pred
+
+                        candidate.split_info = thresh
+
+            if candidate.merit.loss_mean < best_split.merit.loss_mean:
+                best_split = candidate
+
+        return best_split
+
+    def apply_split(self, split, p_node, p_branch, sgt):
+        # Null split: update tree prediction and reset learning node
+        if split.feature is None:
+            self._prediction += split.merit.delta_pred
+            sgt._n_node_updates += 1
+            self.reset()
+            return
+
+        sgt._n_splits += 1
+        sgt._split_features.add(split.feature)
+
+        branch = (
+            NumericBinaryBranch if split.numerical_feature else NominalMultiwayBranch
+        )
+        child_depth = self.depth + 1
+        leaves = tuple(
+            SGTLeaf(
+                self._prediction + delta_pred,
+                depth=child_depth,
+                split_params=self.split_params.copy(),
+            )
+            for delta_pred in split.merit.delta_pred.values()
+        )
+
+        new_split = split.assemble(
+            branch, self.split_params.copy(), self.depth, *leaves
+        )
+
+        if p_branch is None:
+            sgt._root = new_split
+        else:
+            p_node.children[p_branch] = new_split
+
+    @property
+    def total_weight(self) -> float:
+        return self._update_stats.total_weight
+
+    @property
+    def update_stats(self):
+        return self._update_stats
+
+    @staticmethod
+    def delta_prediction(gh: GradHess, lambda_value: float):
+        # Add small constant value to avoid division by zero
+        return -gh.gradient / (gh.hessian + sys.float_info.min + lambda_value)
+
+    def __repr__(self):
+        return f"{self.prediction()}"
```

Comment:
You can just do `return str(self.prediction())`

Reply:
Done!